core/templates/services/svg-sanitizer.service.ts

```diff
@@ -0,0 +1,179 @@
+  constructor(private sanitizer: DomSanitizer) {}
+
+  cleanMathExpressionSvgString(svgString: string): string {
+    // We need to modify/remove unnecessary attributes added by mathjax
+    // from the svg tag.
+    let domParser = new DOMParser();
+    let doc = domParser.parseFromString(svgString, 'image/svg+xml');
+    doc.querySelectorAll('*').forEach((node) => {
+      if (node.tagName.toLowerCase() === 'svg') {
+        node.removeAttribute('xmlns:xlink');
+        node.removeAttribute('role');
+        // We are removing this attribute, because currently it is not in
+        // the white list of valid attributes.
+        node.removeAttribute('aria-hidden');
+        node.setAttribute('xmlns', 'http://www.w3.org/2000/svg');
+      }
+      // Remove the custom data attributes added by MathJax.
+      // These custom attributes don't affect the rendering of the SVGs,
+      // and they are not present in the white list for allowed attributes.
+      for (let i = 0; i < node.attributes.length; i++) {
+        if (node.attributes[i].name.toLowerCase().startsWith('data-')) {
+          node.removeAttribute(node.attributes[i].name.toLowerCase());
+        }
+      }
+    });
+    return doc.documentElement.outerHTML;
+  }
+
+  extractDimensionsFromMathExpressionSvgString(svgString: string): {
+    height: string, width: string, verticalPadding: string} {
+    // The method below extracts the dimensions from the attributes of a
+    // math SVG string generated by mathJax.
+    let domParser = new DOMParser();
+    let dimensions = {
+      height: '',
+      width: '',
+      verticalPadding: ''
+    };
+    let doc = domParser.parseFromString(svgString, 'image/svg+xml');
+    doc.querySelectorAll('*').forEach((node) => {
+      // Mathjax SVGs have relative dimensions in the unit of 'ex' rather
+      // than 'px'(pixels). Hence the dimensions have decimal points in them,
+      // we need to replace these decimals with a letter so that it's easier
+      // to process and validate the filenames.
+      if (node.tagName.toLowerCase() === 'svg') {
+        dimensions.height = (
+          (node.getAttribute('height').match(/\d+\.*\d*/g)[0]).replace(
+            '.', 'd'));
+        dimensions.width = (
+          (node.getAttribute('width').match(/\d+\.*\d*/g)[0]).replace(
+            '.', 'd'));
+        // This attribute is useful for the vertical alignment of the
+        // Math SVG while displaying inline with other text.
+        // Math SVGs don't necessarily have a vertical alignment, in that
+        // case we assign it zero.
+        let styleValue = node.getAttribute('style').match(/\d+\.*\d*/g);
+        if (styleValue) {
+          dimensions.verticalPadding = styleValue[0].replace('.', 'd');
+        } else {
+          dimensions.verticalPadding = '0';
+        }
+      }
+    });
+    return dimensions;
+  }
+
+  getInvalidSvgTagsAndAttrs(svg: Document): {tags: string[], attrs: string[]} {
```

Comment:
This and the next method should probably be private and not exposed in the API.

External callers should only use something like isValidBase64Svg(). Perhaps modify that to raise/return the appropriate error messages.

Reply:
I don't think I understand the changes expected here. My dilemma is if change the function `isValidBase64Svg()` to return error messages, then the name won't remain the same. If I try to change the name, we are trying to get invalid tags and attributes which brings us back to this problem. For now, I have renamed this to `_getInvalidSvgTagsAndAttrs` and made a new public function `getInvalidSvgTagsAndAttrs` that calls the `_getInvalidSvgTagsAndAttrs` function

Reply:
As per our offline discussion, I haven't made the change requested. But I did make some new changes as mentioned in the comment above. PTAL!